### ISSUE TITLE
RFC for Workflow Enhancement: Managing Customer-Specific Product Reservations and Gradual Deliveries in Odoo

### ISSUE THREAD

**Issue by rafaelbn** [OCA/stock-logistics-workflow#1502]
### Summary:

This RFC proposes the introduction of a comprehensive workflow within Odoo to manage customer-specific product reservations, allowing for invoiced merchandise to be held and its delivery to be fractioned over time. This feature aims to address the lack of dedicated support for handling goods reserved for customers, providing a clear distinction from supplier-related functionalities.

### Background:

Businesses often face scenarios where customers purchase goods but request the seller to store these items for future, gradual delivery. This practice requires meticulous management of reserved stock, ensuring it is not mistakenly sold to others and that it can be efficiently delivered in parts as requested by the customer. Currently, Odoo lacks a direct method to facilitate this process, leading to potential inventory mismanagement and customer service challenges.

### Key Requirements:

1. Customer-Specific Reservations: Implement functionality to mark inventory as reserved for a specific customer upon sale, ensuring these items are not available for sale to others.
2. Visibility and Management: Enhance user interfaces to allow for easy management and visibility of customer-specific reservations within sales and inventory modules.
3. Gradual Delivery Support: Integrate a workflow that supports the partial delivery of reserved goods, tracking each delivery's impact on the reserved inventory and updating the order status accordingly.
4. Inventory Adjustment and Alerts: Ensure that reserved goods are reflected accurately in inventory levels and provide alerts during sales order creation if reserved stock can fulfill the order.
5. Detailed Reporting: Develop reporting capabilities to offer insights into the status of all customer reservations, including remaining inventory, delivered quantities, and pending deliveries.

### Proposed Solution:

The solution involves creating or enhancing a module that ties into Odoo's existing sales and inventory management systems. This module would introduce:

- New models or extensions to existing models for handling customer reservations and tracking partial deliveries.
- User interface improvements for managing and viewing reserved goods, including integration with sales orders and inventory reporting.
- Logic to adjust inventory levels based on reservations and deliveries, ensuring accurate stock counts and preventing sales conflicts.
- Notification systems to alert users about potential stock allocations for new sales orders based on existing reservations.
- Comprehensive reporting tools to provide detailed overviews of reservation statuses, facilitating better decision-making and customer communication.

### Business Impact:

Adopting this workflow will significantly enhance inventory accuracy, improve customer satisfaction by honoring their logistical preferences, and optimize stock management by ensuring reserved goods are allocated correctly and efficiently.

### Conclusion:

By addressing the need for a dedicated workflow to manage customer-specific reservations and gradual deliveries, this RFC seeks to fill a critical gap in Odoo's capabilities. We invite the community's feedback, suggestions, and development contributions to realize this important enhancement, reinforcing Odoo's commitment to meeting diverse business needs.

@moduon MT-3404 @EmilioPascual already working in the solution. @rousseldenis if you are interested ping me please.

**Comment by rousseldenis** [comment]
@rafaelbn have you ever looked at wms repo?

**Comment by rafaelbn** [comment]
Hello @rousseldenis , yes we did.

About this matter, I asked Joel in OCA Days, he didn't point me to a solution. And now checking it, I don't see any module or module that could solve the problem.

About wms repo: We ofter go there to look for, but then we find a module that dependes in other module that depends in other modules that depends in other module so the complexity to understand the solution for a specific problem is so complex. I understand modularity but sometimes the dependencies makes we have to use something we don't need.

About this issue The simple solution we found was https://github.com/Tecnativa/extra-addons/tree/16.0/sale_stock_deposit_mgmt and after testing, and testing, reviewing, improving readme, etc... it doesn't convince us.

Thank you for answering! Really appreciated.
